models/TransactionSummary.go

```diff
@@ -0,0 +1,15 @@
+package models
+
+/*
+TransactionSummary models entries to the TransactionSummary table
+*/
+type TransactionSummary struct {
+	UserId        uint32  `gorm:"primary_key;column:userId;not null"`
+	StockId       uint32  `gorm:"primary_key;column:stockId;not null"`
+	StockQuantity int64   `gorm:"column:stockQuantity;not null"`
+	Price         float64 `gorm:"column:price;not null"`
```

Comment:
Won't it be simpler to have this is uint32/64? It won't make much of a difference to us, as price will typically be high enough to not make 0.xy much of a difference.

Haven't thought this through, but just maybe consider if integers are sufficient. Floats can cause unexpected issues, ints are easier to reason with.

Reply:
No it has to be float because the value we're storing in the database is a float (since it's effectively an average price the stock was bought/short sold at). Making it an int will truncate it which, if the number of stocks are high enough (which is very well possible), will make a significant difference.